backend/sirius-web-emf/src/main/java/org/eclipse/sirius/web/emf/view/ViewConverter.java

```diff
@@ -160,18 +168,29 @@ private NodeDescription convert(org.eclipse.sirius.web.view.NodeDescription view
         if (viewNodeDescription.getCreationMode() == Mode.AUTO) {
             synchronizationPolicy = SynchronizationPolicy.SYNCHRONIZED;
         }
+        final String nodeType;
+        if (viewNodeDescription.getStyle().getShape() == null) {
+            nodeType = NodeType.NODE_RECTANGLE;
+        } else {
+            nodeType = NodeType.NODE_IMAGE;
+        }
         // @formatter:off
         NodeDescription result = NodeDescription.newNodeDescription(this.idProvider.apply(viewNodeDescription))
                 .targetObjectIdProvider(this.semanticTargetIdProvider)
                 .targetObjectKindProvider(this.semanticTargetKindProvider)
                 .targetObjectLabelProvider(this.semanticTargetLabelProvider)
                 .semanticElementsProvider(this.getSemanticElementsProvider(viewNodeDescription))
                 .synchronizationPolicy(synchronizationPolicy)
-                .typeProvider(variableManager -> NodeType.NODE_RECTANGLE)
+                .typeProvider(variableManager -> nodeType)
                 .labelDescription(this.getLabelDescription(viewNodeDescription))
-                .styleProvider(variableManager -> this.stylesFactory.createNodeStyle(viewNodeDescription.getStyle().getColor()))
+                .styleProvider(variableManager -> {
+                    String shapeId = viewNodeDescription.getStyle().getShape();
+                    String shapeFileName = this.customImagesService.findById(UUID.fromString(shapeId)).map(CustomImage::getFileName).orElse(DEFAULT_SHAPE_FILE);
```

Comment:
I'm still puzzled with this id + filename identity. Why aren't you using the id here? The id could be used as the sole identity of the image everywhere without using the filename at all. Since it's not displayed to the end user, we could remove it entirely.

Reply:
I know, the current design is problematic.
I added a `id UUID` because it's the convention we follow, and I guess it's needed by Hibernate? Not sure.
But I also needed _something_ to be able to determine the file type (and thus which `MediaType` to return in `ImageController`), hence the `fileName` (yes, just the image type/extension would be enough).

Reply:
Another issue with using the id is that they are randomly generated (at least how we do it in other entities), and this means re-importing images breaks any existing diagrams that refer to the id in some of hteir nodes' styles.
This would also make such diagrams non-portable between servers (upload + download) even if they have the "same" custom images imported, as they would have different ids on different instances.